ecs-agent/netlib/platform/common_linux.go

```diff
@@ -205,3 +259,352 @@ func (c *common) buildSingleNSNetConfig(
 		proxyConfig,
 		ifaces...)
 }
+
+// CreateNetNS creates a new network namespace with the specified path.
+func (c *common) CreateNetNS(netNSPath string) error {
+	nsExists, err := c.nsUtil.NSExists(netNSPath)
+	if err != nil {
+		return errors.Wrapf(err, "failed to check netns %s", netNSPath)
+	}
+
+	if nsExists {
+		return nil
+	}
+
+	err = c.nsUtil.NewNetNS(netNSPath)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create netns %s", netNSPath)
+	}
+
+	// The loopback interface in a new network namespace is down by default in Linux.
+	// In case of a container launched using Docker, Docker itself ensures that loopback is up.
+	// Manually set the operational state to up to allow loopback communication.
+	err = c.nsUtil.ExecInNSPath(netNSPath, c.setUpLoFunc(netNSPath))
+
+	return err
+}
+
+// setUpLoFunc returns a method that sets the loop back interface inside a
+// particular network namespace to the state "UP". This function is used to
+// set up the loop back interface inside a task network namespace soon after
+// its creation.
+func (c *common) setUpLoFunc(netNSPath string) func(cnins.NetNS) error {
+	return func(cnins.NetNS) error {
+		// Get a handle to the loop back interface.
+		link, err := c.netlink.LinkByName("lo")
+		if err != nil {
+			return errors.Wrapf(err, "failed to find loopback interface in %s", netNSPath)
+		}
+
+		// Bring up the interface (ip link set dev lo up).
+		err = c.netlink.LinkSetUp(link)
+		if err != nil {
+			return errors.Wrapf(err, "failed to bring up loopback interface in %s", netNSPath)
+		}
+
+		return nil
+	}
+}
+
+// createDNSConfig creates the DNS config files for a particular network namespace.
+// If the agent is running on debug mode, it reuses the host's DNS config files.
+// If not, it gathers the DNS related data from the netns primary interface.
+// The DNS files are written to the network namespace dir "/etc/netns/<netns-name>/".
+// Afterward, these files are copied into the task volume to be bind-mounted into
+// task containers.
+func (c *common) createDNSConfig(
+	taskID string,
+	reuseHostDNSConfig bool,
+	netNS *tasknetworkconfig.NetworkNamespace) error {
+	// For debug mode, resolv.conf and hosts files are same as the host machine.
```

Comment:
Where/how do we use debug mode?

Reply:
This is a containerd agent specific mode where we run Fargate owned AMIs on EC2 launch type.